src/main/kotlin/org/rust/ide/inspections/RsConstOrTypeParamsInExternInspection.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsTypeParameterList
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.RsAbstractableOwner
+import org.rust.lang.core.psi.ext.owner
+import org.rust.lang.utils.RsDiagnostic
+import org.rust.lang.utils.addToHolder
+
+class RsConstOrTypeParamsInExternInspection : RsLocalInspectionTool() {
```

Comment:
We don't want to create a separate inspection for most (ideally, for all) compiler errors because inspection can be disabled by a user. But you shouldn't be able to disable compiler error.
Let's use `Annotator` API. Since this error is syntax only, I suggest `RsSyntaxErrorsAnnotator`